### PR TITLE
Rework skip-binners

### PR DESCRIPTION
**Pull request by AroneyS**
Rework skip-binners to use optional inputs in dastool rule. Avoids complex file/folder generation.
Could be extended to make refinement optional for each binner.

Also includes some logic-testing of the snakemake pipeline. Costly tests since they run a shell, but at least they don't run the actually commands.

### CODE REVIEW

**Comment by rhysnewell** [comment]
Hey @AroneyS ,

Looks good, but can you verify that in the dastool rule that the `Fasta_to_Scaffolds2Bin.sh` doesn't error out when you point it to a directory that doesn't exist since the output folders for skipped binners won't be created with your method.

https://github.com/rhysnewell/aviary/blob/32b70973616e8279bfd998182eb31b71652ceda3/aviary/modules/binning/binning.smk#L560-L605

File: test/test_integration.py [hunk 1]
```diff
@@ -22,36 +22,62 @@
...
 import unittest
-import sys
-import subprocess
 import tempfile
 import os.path
-
 import extern
 
 data = os.path.join(os.path.dirname(__file__), 'data')
+conda = os.path.join(data,'.conda')
```
Comment (wwood) [inline test/test_integration.py:30]: unused? no biggie

**Comment by AroneyS** [comment]
> Hey @AroneyS ,
> 
> Looks good, but can you verify that in the dastool rule that the `Fasta_to_Scaffolds2Bin.sh` doesn't error out when you point it to a directory that doesn't exist since the output folders for skipped binners won't be created with your method.
> 
> https://github.com/rhysnewell/aviary/blob/32b70973616e8279bfd998182eb31b71652ceda3/aviary/modules/binning/binning.smk#L560-L605

Yes. It gives the same error as when there are no bins produced by the binner. Since the commands are joined by `;`, it just continues.

**Review by wwood — APPROVED** [review]

**Comment by rhysnewell** [comment]
Hey @wwood , 

I'd prefer if merges went into `dev` and then `dev` gets merged into `main` when a new release is made

**Comment by wwood** [comment]
Ah sure, I thought about reverting but think that might be too annoying to admin - go ahead if you want.

**Comment by rhysnewell** [comment]
Nah, just in future something to be aware of

**Pull request merged**